File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/buildings/autorisations/TypeAutorisation.java
package io.github.alexiscomete.lapinousecond.worlds.buildings.autorisations;

import io.github.alexiscomete.lapinousecond.entity.Owner;

public class TypeAutorisation<U> implements BuildingAutorisation {

    final Class<U> uClass;

    public TypeAutorisation(Class<U> uClass) {
        this.uClass = uClass;
    }

    @Override
    public boolean isAutorise(Owner owner) {
        return uClass.isInstance(owner);
    }
}

File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/buildings/autorisations/TypeAutorisation.kt
package io.github.alexiscomete.lapinousecond.worlds.buildings.autorisations

import io.github.alexiscomete.lapinousecond.entity.entities.Owner

class TypeAutorisation<U>(private val uClass: Class<U>) : BuildingAutorisation {
    override fun isAutorise(owner: Owner?): Boolean {
        return uClass.isInstance(owner)
    }
}